src/consensus/block_producer.rs

```diff
@@ -649,4 +662,110 @@ mod tests {
         assert_eq!(slot, ret.0);
         assert_eq!(block_info.hash, ret.1);
     }
+
+    #[tokio::test]
+    async fn verify_produce_block_parent_not_ready() {
+        let slot = Slot::windows().nth(10).unwrap();
+        let block_info = BlockInfo {
+            hash: [1; 32],
+            parent: (slot.prev(), [2; 32]),
+        };
+
+        let (first_slice_finished_tx, first_slice_finished_rx) = oneshot::channel();
+        let (start_second_slice_tx, start_second_slice_rx) = oneshot::channel();
+
+        let mut seq = Sequence::new();
+        let mut blockstore = MockBlockstore::new();
```

Comment:
This use of the MockBlockstore looks more complicated than doing the same for the actual blockstore. Or ist there some detail in this setup that I might be missing?

Reply:
No, I agree, this block of code is too complicated.  I wanted to checkpoint this PR and attempt to reproduce the test using the actual blockstore to see what that takes.